src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ValidateVariants.java

```diff
@@ -367,50 +449,185 @@ private void validateGVCFVariant(final VariantContext vc) {
         // Note: VariantContext.validateRSIDs blows up on an empty list (but works fine with null).
         // The workaround is to not pass an empty list.
-        switch( t ) {
-            case ALL:
-                if(hasReference())
-                {
-                    if(!rsIDs.isEmpty())
-                    {
-                        vc.extraStrictValidation(reportedRefAllele, observedRefAllele, rsIDs);
-                    }
-                    else{
-                        vc.validateReferenceBases(reportedRefAllele, observedRefAllele);
-                        vc.validateAlternateAlleles();
-                        vc.validateChromosomeCounts();
-                    }
-                }
-                else{
-                    if (rsIDs.isEmpty())
-                    {
-                        vc.validateAlternateAlleles();
-                        vc.validateChromosomeCounts();
-                    }
-                    else{
-                        vc.validateAlternateAlleles();
-                        vc.validateChromosomeCounts();
-                        vc.validateRSIDs(rsIDs);
-                    }
-                }
-                break;
-            case REF:
-                vc.validateReferenceBases(reportedRefAllele, observedRefAllele);
-                break;
-            case IDS:
-                if (!rsIDs.isEmpty()) {
-                    vc.validateRSIDs(rsIDs);
-                }
-                break;
-            case ALLELES:
-                vc.validateAlternateAlleles();
-                break;
-            case CHR_COUNTS:
-                vc.validateChromosomeCounts();
+       if(validationsToPerform[ValidationType.REF.ordinal()]) {
+           if (hasReference()) {
+               currentType = ValidationType.REF;
+               vc.validateReferenceBases(reportedRefAllele, observedRefAllele);
+           }
+       }
+       if (validationsToPerform[ValidationType.IDS.ordinal()]) {
+           if (!rsIDs.isEmpty()) {
+               currentType = ValidationType.IDS;
+               vc.validateRSIDs(rsIDs);
+           }
+       }
+       if (validationsToPerform[ValidationType.ALLELES.ordinal()]) {
+           currentType = ValidationType.ALLELES;
+           vc.validateAlternateAlleles();
+       }
+       if (validationsToPerform[ValidationType.CHR_COUNTS.ordinal()]) {
+           currentType = ValidationType.CHR_COUNTS;
+           vc.validateChromosomeCounts();
+       }
+       if (validationsToPerform[ValidationType.GNARLY_INPUT.ordinal()]) {
+           currentType = ValidationType.GNARLY_INPUT;
+           validateGnarlyAnnotations(vc);
+           validateGnarlyGenotypes(vc);
+       }
+       if (validationsToPerform[ValidationType.VQSR_INPUT.ordinal()]) {
+           currentType = ValidationType.VQSR_INPUT;
+           if (vc.isReferenceBlock()) {
+               return;
```

Comment:
Are you sure returning is right here? Not breaking?

Reply:
This implies it is automatically skipping AS_ANNOTATIONS if its a reference block AND `VQSR_INPUT` is set but not if `VQSR_INPUT` is excluded.